test/test_fx.py

```diff
@@ -600,6 +600,22 @@ def test_construct_root_dict(self):
         ref_out : torch.Tensor = linear_mod(x) + add_param
         self.assertEqual(out, ref_out)
 
+    def test_symbolic_trace_assert(self):
+        message = "assert_foobar"
+
+        class AssertsTensorShape(torch.nn.Module):
+            def forward(self, x):
+                torch.do_assert(x.shape[1] > 4, message)
```

Comment:
Is there a way to make `torch.do_assert` work in non-symbolic-tracing mode as well? What we want to avoid is having to branch based on whether we are in symbolic tracing mode:
```python
if is_symbolic_tracing():
  torch.do_assert(x.shape[1] > 4, message)
else:
  assert x.shape[1] > 4, message
```
As we use assertions extensively in DPER3 code, having to add branches similar to this will make the code much harder to maintain.

Reply:
yes, it works both traced and non-traced without needing to branch.  I'll add a test case.